x-pack/plugins/observability_solution/inventory/public/components/alerts_badge/alerts_badge.test.tsx

```diff
@@ -78,11 +86,40 @@ describe('AlertsBadge', () => {
 
       alertsCount: 5,
     };
+    mockAsKqlFilter.mockReturnValue('service.name: bar');
 
     render(<AlertsBadge entity={entity} />);
     expect(screen.queryByTestId('inventoryAlertsBadgeLink')?.getAttribute('href')).toEqual(
-      `/app/observability/alerts?_a=(kuery:\"entity1\",status:active)`
+      "/app/observability/alerts?_a=(kuery:'service.name: bar',status:active)"
     );
     expect(screen.queryByTestId('inventoryAlertsBadgeLink')?.textContent).toEqual('5');
   });
+  it('render alerts badge for a service entity with multiple identity fields', () => {
+    const entity: InventoryEntity = {
+      ...(commonEntityFields as InventoryEntity),
+      entityType: 'service',
+      entityDisplayName: 'foo',
+      entityIdentityFields: ['service.name', 'service.environment'],
+      entityDefinitionId: 'service',
+      service: {
+        name: 'bar',
+        environment: 'prod',
+      },
+      agent: {
+        name: 'node',
+      },
+      cloud: {
+        provider: null,
+      },
+      alertsCount: 2,
+    };
+
+    mockAsKqlFilter.mockReturnValue('service.name: bar AND service.environment: prod');
+
+    render(<AlertsBadge entity={entity} />);
+    expect(screen.queryByTestId('inventoryAlertsBadgeLink')?.getAttribute('href')).toEqual(
+      "/app/observability/alerts?_a=(kuery:'service.name: bar AND service.environment: prod',status:active)"
```

Comment:
For the `service.name`, agents don't allow illegal character like `:` or `?` .  For the environment though you can set up `my:environment`. To ensure the search works correctly, can we please use double quotes around the values?

Reply:
Good catch! `asKqlFilter` will return double quotes normally and the test was there before (I just reverted it and haven't looked in detail) so I guess it was the mock that was created without them so I added them here and as a mock value :)

Reply:
Interesting that `toEqual` didn't catch the missing quotes. Thanks for checking it